src/rgw/rgw_data_sync.cc

```diff
@@ -3152,8 +3155,9 @@ class InitBucketFullSyncStatusCR : public RGWCoroutine {
 
       if (info.oldest_gen == 0) {
         if (check_compat) {
+          assert(info.latest_gen == 0);
```

Comment:
i don't think this is right. the case we're worried about for `check_compat` is after upgrading from a ceph version that didn't track the bucket-wide status (or support multisite resharding). after the upgrade, you can enable resharding and reshard this bucket. and in the absense of other writes, a reshard may be the first thing that writes datalog entries for this bucket to trigger our sync here